### PR TITLE
docs: migrating to GitHub Pages

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
Migrating docs from readthedocs to GitHub pages.

The existing readthedocs website will be deleted after couple of weeks.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+site_name: Pytest Splunk Addon
+site_author: Splunk
+site_url: "https://splunk.github.io/pytest-splunk-addon/"
+edit_uri: "tree/migrating-docs/docs/"
```
Comment (artemrys) [inline mkdocs.yml:19]:
```suggestion
edit_uri: "tree/main/docs/"
```

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  features:
+    - content.code.copy
+    - navigation.indexes
+    - navigation.expand
+
+extra_css:
+    - css/extra.css
+
+nav:
+  - Home: "overview.md"
```
Comment (artemrys) [inline mkdocs.yml:62]:
```suggestion
  - Home: "index.md"
```

Please rename `overview.md` to `index.md` so when the website opens it does not show 404.

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+Or, in any case, if pip is unavailable:
+
+```console
+1. git clone https://github.com/splunk/pytest-splunk-addon.git
+2. cd pytest-splunk-addon
+3. pip install poetry
+4. poetry install
+```
```
Comment (artemrys) [inline docs/overview.md:37]: Let's remove this.

File: docs/how_to_use.md [hunk 4]
```diff
@@ -0,0 +1,453 @@
+# How To Use
+
+
+Create a test file in the tests folder
+
+```python
+from pytest_splunk_addon.standard_lib.addon_basic import Basic
```
Comment (artemrys) [inline docs/how_to_use.md:7]:
```suggestion
from pytest_splunk_addon.standard_lib.addon_basic import Basic


```

File: docs/how_to_use.md [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+
+## Test Execution
+
+There are three ways to execute the tests:
+
+**1. Running tests with an external Splunk instance**
+
+ ```python3
+ pip3 install pytest-splunk-addon
+ ```
```
Comment (artemrys) [inline docs/how_to_use.md:21]: Let's assume we already installed PSA - let's remove this.

File: docs/how_to_use.md [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+
+**2. Running tests with docker splunk**
+
+ ```bash
+ git clone git@github.com:splunk/pytest-splunk-addon.git
+ cd pytest-splunk-addon
+ pip install poetry
+ poetry install
+ ```
```
Comment (artemrys) [inline docs/how_to_use.md:36]: Let's remove this one as well.

File: docs/how_to_use.md [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+ ARG SPLUNK_APP_ID=TA_UNKNOWN
+ ARG SPLUNK_APP_PACKAGE=$SPLUNK_APP_PACKAGE
+ COPY $SPLUNK_APP_PACKAGE /opt/splunkforwarder/etc/apps/$SPLUNK_APP_ID
+ ```
+
+<details>
+<summary>Create docker-compose.yml</summary>
+
+```commandline
+version: "3.7"
```
Comment (artemrys) [inline docs/how_to_use.md:65]: Is there a way to source the content of this file from the repository itself? Same for below file.

**Review by artemrys — CHANGES_REQUESTED** [review]

File: mkdocs.yml (hunk 1 above)
Comment (harshilgajera-crest) [reply mkdocs.yml]: done

File: mkdocs.yml (hunk 2 above)
Comment (harshilgajera-crest) [reply mkdocs.yml]: done

File: docs/how_to_use.md (hunk 6 above)
Comment (harshilgajera-crest) [reply docs/how_to_use.md]: done

File: docs/how_to_use.md (hunk 7 above)
Comment (artemrys) [reply docs/how_to_use.md:56]: Discussed in Slack - let's leave it as it is right now.

**Review by artemrys — APPROVED** [review]

**Pull request merged**